src/main/java/org/jvnet/hudson/update_center/GitHubSource.java

```diff
@@ -70,6 +82,110 @@ private void retrieveRepositoryNames() throws IOException {
         Files.write(GITHUB_REPO_LIST.toPath(), ret);
     }
 
+    public Map<String, List<String>> getTopics(String organization) throws IOException {
+        if (this.topicNames != null) {
+            return this.topicNames;
+        }
+        System.err.println("Retrieving GitHub topics...");
+        Cache cache = new Cache(GITHUB_API_CACHE, 20L * 1024 * 1024); // 20 MB cache
+        OkHttpClient client = new OkHttpClient.Builder()
+                .cache(cache)
+                .authenticator(new Authenticator() {
+                    @Nullable
+                    @Override
+                    public Request authenticate(Route route, Response response) throws IOException {
+                        String credential = Credentials.basic(GITHUB_API_USERNAME, GITHUB_API_PASSWORD);
+                        return response.request().newBuilder().header("Authorization", credential).build();
+                    }
+                }).build();
+
+        boolean hasNextPage = true;
+        String endCursor = null;
+
+        Map<String,List<String>> ret = new HashMap<>();
+
+        while (hasNextPage) {
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("query", String.format("{\n" +
```

Comment:
maybe read this from a resource file? so you can read it a lot better

Reply:
Is that a thing you normally do in java? I always like having all the code together but i can stick to conventions

Reply:
bit of a mix really, 
in a unit test with something like the a json response you would almost always put it in a file so that you get syntax  checking and can easily copy and paste it into something else

with something like SQL you would often have it directly in the code.

main benefit of moving it would be you can copy it out and run it directly, and also you could validate the query with a syntax checker